osu.Game/Database/LegacyExporter.cs

```diff
@@ -33,17 +33,17 @@ protected LegacyExporter(Storage storage)
             UserFileStorage = storage.GetStorageForDirectory(@"files");
         }
 
+        protected virtual string GetItemExportString(TModel item) => item.GetDisplayString().GetValidFilename();
+
         /// <summary>
         /// Exports an item to a legacy (.zip based) package.
         /// </summary>
         /// <param name="item">The item to export.</param>
         public void Export(TModel item)
         {
-            string itemFilename = item.GetDisplayString().GetValidFilename();
-
-            IEnumerable<string> existingExports = exportStorage.GetFiles("", $"{itemFilename}*{FileExtension}");
+            IEnumerable<string> existingExports = exportStorage.GetFiles("", $"{GetItemExportString(item)}*{FileExtension}");
 
-            string filename = NamingUtils.GetNextBestFilename(existingExports, $"{itemFilename}{FileExtension}");
+            string filename = NamingUtils.GetNextBestFilename(existingExports, $"{GetItemExportString(item)}{FileExtension}");
```

Comment:
Always save a method result to a local rather than evaluating it every time, i.e.:
```csharp
string filename = GetItemExportString(item).GetValidFilename();

IEnumerable<string> existingExports = exportStorage.GetFiles("", $"{filename}*{FileExtension}");

filename = NamingUtils.GetNextBestFilename(existingExports, $"{filename}{FileExtension}");

Reply:
Is this for code style and readability?

Reply:
More of a rule of thumb. Can save performance in some cases where the method comes with an overhead (e.g. a database query).